packages/eyeglass/src/util/typescriptUtils.ts

```diff
@@ -0,0 +1,7 @@
+export interface Dict<T> {
+  [key: string]: T;
+}
+
+export function isPresent<T>(v: T): v is Exclude<T, undefined | null> {
```

Comment:
IIRC this implementation keeps downstream types simpler (and possibly more performant) since it doesn't involve use of a mapped type, and doesn't result in `Exclude<K,T>` trickling through the code paths that use this function

```ts

export function isPresent<T>(v: T | undefined | null): v is T {
```

Reply:
I thought this would have the incorrect behavior with an explicit value provided for `T` but it doesn't. I like this better! Fixed in f9132636164144049b6facc1413a7a02909790be.